src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -173,6 +173,15 @@ fn compile<'a, 'cfg: 'a>(cx: &mut Context<'a, 'cfg>,
         return Ok(())
     }
 
+    // REVIEW moving this code changes the dependency traversal from pre- to
```

Comment:
I think this should be ok, yeah.

Although I'd prefer for it to not be necessary, I'll comment below.